qiskit/quantum_info/operators/symplectic/pauli_list.py

```diff
@@ -1062,3 +1062,37 @@ def from_symplectic(cls, z, x, phase=0):
         """
         base_z, base_x, base_phase = cls._from_array(z, x, phase)
         return cls(BasePauli(base_z, base_x, base_phase))
+
+    def _qw_anticommutation_graph(self):
+        """Create qubit wise anti commutation graph with edges (i, j) if i and j are not commutable.
+
+        Returns:
+            List[Tuple(int,int)]: A list of pairs of indices of the PauliList that are not commutable.
+        """
+        anti_commutation_graph = []
+        for i in range(self._num_paulis):
+            for j in range(i, self._num_paulis):
+                if i != j and self[i].qw_anticommutes(self[j]):
+                    anti_commutation_graph.append((i, j))
+        return anti_commutation_graph
+
+    def group_subops_pauli_list(self):
+        """Partition a  PauliList into sets of mutually qubit-wise commuting Pauli strings.
+
+        Returns:
+            List[PauliList]: List of PauliLists where each PauliList contains commutable Pauli operators.
+        """
+        import retworkx as rx
+        from collections import defaultdict
```

Comment:
Is there a reason to have the imports at runtime? This adds overhead to every function call. If there isn't an issue with circular imports (which I'd be really surprised here if there was because they're external to terra) we should use module level imports.

Reply:
There wasn't a particular reason, I'll change it to use module level imports.